incubator/honeycomb-agent/README.md

```diff
@@ -0,0 +1,144 @@
+# Honeycomb Mixin
+> The [Honeycomb Kubernetes agent](https://github.com/honeycombio/honeycomb-kubernetes-agent/tree/master) is a monitoring tool for pods and their containerized applications (e.g. nginx). It continuously observes, parses, and sends their log files to the Honeycomb API, where the data is rendered in a dashboard for real-time observability.
+
+## Overview
```

Comment:
Just to track this, @abiogenesis-now and I chatted and agreed that it's important for the user to understand that:

1. `honeycomb-agent.libsonnet` is a library that users can use to add the Honeycomb agent to their applications.
1. The `examples` directory contains some samples of how to use this library.

Later, and not necessarily in this section, it's probably important that they also understand:

1. The core abstractions in the `app::` namespace (which currently only allows you to build a `DaemonSet`)
1. That eventually this library will also support adding the agent to (_e.g._) a `Deployment`.